torch/distributed/device_mesh.py

```diff
@@ -451,21 +451,67 @@ def get_group(
                 return dim_groups
 
         @staticmethod
-        def from_group(group: ProcessGroup, device_type: str) -> "DeviceMesh":
+        def from_group(
+            group: Union[ProcessGroup, List[ProcessGroup]],
+            device_type: str,
+            mesh: Optional[Union[torch.Tensor, "ArrayLike"]] = None,
+            *,
+            mesh_dim_names: Optional[Tuple[str, ...]] = None,
+        ) -> "DeviceMesh":
             """
             Contstructs a :class:`DeviceMesh` with ``device_type`` from an
             existing :class:`ProcessGroup`.
 
-            The constructed device mesh is assumed to be 1D.
+            The constructed device mesh has number of dimensions equal to the
```

Comment:
is there any requirement for the input pg groups, 
The test case `test_2d_process_group_init ` first builds a global pg, then builds dp_shard_group, dp_replicate_group based on the global pg, is this required to always build global pg first ? 
Or can these two process groups be created independently ?

Reply:
I think the two process groups can be constructed independently.